engine/src/test/java/com/redhat/cloud/notifications/routers/EndpointTestResourceTest.java

```diff
@@ -0,0 +1,125 @@
+package com.redhat.cloud.notifications.routers;
+
+import com.redhat.cloud.notifications.Json;
+import com.redhat.cloud.notifications.TestLifecycleManager;
+import com.redhat.cloud.notifications.db.ResourceHelpers;
+import com.redhat.cloud.notifications.events.FromCamelHistoryFiller;
+import com.redhat.cloud.notifications.ingress.Action;
+import com.redhat.cloud.notifications.ingress.Context;
+import com.redhat.cloud.notifications.ingress.Event;
+import com.redhat.cloud.notifications.ingress.Metadata;
+import com.redhat.cloud.notifications.ingress.Payload;
+import com.redhat.cloud.notifications.models.Endpoint;
+import com.redhat.cloud.notifications.models.EndpointType;
+import com.redhat.cloud.notifications.models.event.TestEventHelper;
+import com.redhat.cloud.notifications.routers.endpoints.EndpointTestRequest;
+import io.quarkus.test.common.QuarkusTestResource;
+import io.quarkus.test.junit.QuarkusTest;
+import io.restassured.http.ContentType;
+import io.smallrye.reactive.messaging.providers.connectors.InMemoryConnector;
+import io.smallrye.reactive.messaging.providers.connectors.InMemorySink;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import javax.enterprise.inject.Any;
+import javax.inject.Inject;
+import java.util.Map;
+
+import static io.restassured.RestAssured.given;
+
+@QuarkusTest
+@QuarkusTestResource(TestLifecycleManager.class)
+public class EndpointTestResourceTest {
+
+    @Any
+    @Inject
+    InMemoryConnector inMemoryConnector;
+
+    @Inject
+    ResourceHelpers resourceHelpers;
+
+    /**
+     * Tests that when a "Test Endpoint" request is received, a "test
+     * endpoint integration" event is posted to the egress channel.
+     */
+    @Test
+    void testEndpoint() {
+        final String orgId = "test-endpoint-engine-test";
+        final Endpoint createdEndpoint = this.resourceHelpers.createEndpoint(EndpointType.CAMEL, "slack", true, 0);
+
+        final EndpointTestRequest endpointTestRequest = new EndpointTestRequest(createdEndpoint.getId(), orgId);
+
+        // Call the endpoint under test.
+        given()
+            .when()
+            .contentType(ContentType.JSON)
+            .body(Json.encode(endpointTestRequest))
+            .post("/internal/endpoints/test")
+            .then()
+            .statusCode(204);
+
+        // We should receive the action triggered by the REST call.
+        InMemorySink<String> actionsOut = this.inMemoryConnector.sink(FromCamelHistoryFiller.EGRESS_CHANNEL);
+        final var actionsList = actionsOut.received();
+
+        // Only one test action should have been sent to Kafka.
+        final var expectedActionsCount = 1;
+        Assertions.assertEquals(expectedActionsCount, actionsList.size(), "unexpected number of actions sent to Kafka");
```

Comment:
Whenever you need to check the messages received by a sink, it is safer for test stability purposes to wait using Awaitility:
https://github.com/RedHatInsights/notifications-backend/blob/41ac032e469c30ab62cb311c7e47f950d4d3a054/engine/src/test/java/com/redhat/cloud/notifications/events/IntegrationDisabledNotifierTest.java#L77

Reply:
Aaah, gotcha! Thanks!